sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAdministrationAsyncClient.java

```diff
@@ -0,0 +1,473 @@
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * The AttestationAdministrationClient provides access to the administrative policy APIs
+ * implemented by the Attestation Service.
+ *
+ * More information on attestation policies can be found <a href='https://docs.microsoft.com/en-us/azure/attestation/basic-concepts#attestation-policy'>here</a>
+ *
+ * There are two main families of APIs available from the Administration client.
+ * <ul>
+ *     <li>Attestation Policy Management</li>
+ *     <li>Attestation Policy Management Certificate Management</li>
+ * </ul>
+ *
+ * Attestation service instances operate in three different modes:
+ * <ul>
+ *     <li>Shared - a shared instance is a regional instance which is available to all customers.
+ *     It does NOT support customer specified policy documents - there is only a default policy available
+ *     for each attestation type </li>
+ *     <li>AAD - An attestation instance where the customer trusts Azure Active Directory (and Azure
+ *     Role Based Access Control) to manage the security of their enclave. </li>
+ *     <li>Isolated - an attestation instance where the customer does *not* trust Azure Active Directory
+ *     (and RBAC) to manage the security of their enclave </li>
+ * </ul>
+ *<p>
+ * When an attestation instance is in Isolated mode, additional proof needs to be provided by the customer
+ * to verify that they are authorized to perform the operation specified.
+ *</p>
+ *
+ */
+@ServiceClient(builder = AttestationClientBuilder.class, isAsync = true)
+public final class AttestationAdministrationAsyncClient {
+    private final SigningCertificatesImpl signingCertificatesImpl;
+    private final PoliciesImpl policyImpl;
+    private final PolicyCertificatesImpl certificatesImpl;
+    private final ClientLogger logger;
+    private final AttestationTokenValidationOptions tokenValidationOptions;
+    private final AtomicReference<List<AttestationSigner>> cachedSigners;
+    private static final SerializerAdapter SERIALIZER_ADAPTER = JacksonAdapter.createDefaultSerializerAdapter();
+
+    /**
+     * Initializes an instance of Attestations client.
+     *
+     * @param clientImpl the service client implementation.
+     */
+    AttestationAdministrationAsyncClient(AttestationClientImpl clientImpl, AttestationTokenValidationOptions tokenValidationOptions) {
+        this.signingCertificatesImpl = clientImpl.getSigningCertificates();
+        this.policyImpl = clientImpl.getPolicies();
+        this.certificatesImpl = clientImpl.getPolicyCertificates();
+        this.tokenValidationOptions = tokenValidationOptions;
+        this.logger = new ClientLogger(AttestationAdministrationAsyncClient.class);
+        this.cachedSigners = new AtomicReference<>(null);
+    }
+//region Get Attestation Policy.
```

Comment:
[nit] Remove?

Reply:
Remove which? The `ClientLogger`? Or the `cachedSigners`? I think I need both of these, either to handle exceptions or to cache the signers used to verify attestation tokens.

Or are you asking about the `//region`? 

Sorry - I sometimes have trouble figuring out the context of comments.

Reply:
Ohh - thanks for letting me know. I think this might be because I use codeflow wrapper. And it is not always able to correctly map. I meant the //region.

Reply:
Is the use of regions discouraged? I know they're not officially supported outside of IntelliJ, but I often find them helpful distinguishing between families of APIs. But I'm totally fine with removing them.

Reply:
Hmm... I am not aware of them being discouraged or anything. It is however the first time I am seeing them being used in the code. If you find them useful keep them (I pointed out a 'nit' since it was commented out).

Reply:
It's not technically commented out in IntelliJ.

Instead, it triggers a scope marker (+), which can be minimized.
![image](https://user-images.githubusercontent.com/8220172/141359050-68118fdf-0727-49eb-9ebc-8ea5476a58a2.png)

<img width="291" alt="image" src="https://user-images.githubusercontent.com/8220172/141359143-047eb3f8-ce1b-4279-b34b-dda63a5faec2.png">

I don't use them often, but with 6 sets of APIs in a single file, I sometimes find it helpful to be able to hide some of them. And regions are extremely [controversial](https://marcduerst.com/2016/10/03/c-regions-are-evil/)

Reply:
TIL, thanks :)